plugin/pkg/scheduler/scheduler_test.go

```diff
@@ -169,163 +168,164 @@ func TestScheduler(t *testing.T) {
 	}
 }
 
-func TestSchedulerForgetAssumedPodAfterDelete(t *testing.T) {
-	// Set up a channel through which we'll funnel log messages from the watcher.
-	// This way, we can guarantee that when the test ends no thread will still be
-	// trying to write to t.Logf (which it would if we handed t.Logf directly to
-	// StartLogging).
-	ch := make(chan string)
-	done := make(chan struct{})
-	var wg sync.WaitGroup
-	wg.Add(1)
+func TestSchedulerNoPhantomPodAfterExpire(t *testing.T) {
+	stop := make(chan struct{})
+	defer close(stop)
+	queuedPodStore := clientcache.NewFIFO(clientcache.MetaNamespaceKeyFunc)
+	scache := schedulercache.New(1*time.Second, stop)
+	scheduler, _, bindingChan, _ := setupTestSchedulerWithOnePod(t, queuedPodStore, scache)
+
+	waitPodExpireChan := make(chan struct{})
+	timeout := make(chan struct{})
 	go func() {
-		defer wg.Done()
 		for {
 			select {
-			case msg := <-ch:
-				t.Log(msg)
-			case <-done:
+			case <-timeout:
 				return
+			default:
 			}
+			pods, err := scache.List(labels.Everything())
+			if err != nil {
+				t.Fatalf("cache.List failed: %v", err)
+			}
+			if len(pods) == 0 {
+				close(waitPodExpireChan)
+				return
+			}
+			time.Sleep(1 * time.Second)
```

Comment:
This is unit test, which is supposed to be pretty fast - please make the sleep period smaller here.